CONTRIBUTING.md

```diff
@@ -0,0 +1,64 @@
+
+1. Search whether your bug or feature was already reported. You can do so on [the GitHub issue tracker of nixos/nix](https://github.com/NixOS/nix/issues).
+
+2. If you were not able to find the bug or feature [open a new issue](https://github.com/NixOS/nix/issues/new/choose)
+
+3. The issue template will guide you in what should be in an issue.
+   The more complete the information, the more likely it can be found by others and the more useful it is in the future.
+   Try to make sure reported bugs can be reproduced by other contributors.
+
+4. Once submitted, do not expect issues to be picked up or solved right away.
+   The only way to ensure this, is to work on the issue yourself. See [Making changes to Nix](#making-changes-to-nix).
+
+## Report a security vulnerability
+
+Check out the [security policy](https://github.com/NixOS/nix/security/policy).
+
+## Making changes to Nix
+
+1. Search for already existing pull requests that might cover the contribution you are about to make.
+   There are many open pull requests that might already do what you intent to work on.
+
+2. Search for related issues that cover what you're going to work on. It could help to mention there that you will work on the issue.
+
+3. Check [the manual](https://nixos.org/manual/nix/stable/contributing/hacking.html) for information on building Nix and running its tests.
+   
+   For contributions on the CLI, please check the manual page on [CLI guidelines](https://nixos.org/manual/nix/stable/contributing/cli-guideline.html).
+   
+4. Make your changes!
+
+5. [Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) for your changes
+   * [Mark the pull request as draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) if you're not done with the changes.
+   * Make sure to have [a clean history of commits on your branch by using rebase](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request).
+   * Link related issues in your pull request to inform interested parties and future contributors about your change.
+
+6. Do not expect your pull request to be reviewed immediately.
+   The core Nix developers are overwhelmed with notifications and reviewing other pull requests.
+
+7. Ask other contributors for feedback and to shepherd your pull request into shape.
+   You can do so by opening a new topic on [Discourse in the Nix Development category](https://discourse.nixos.org/c/dev/nix/50).
+
+## Making changes to the Nix manual
+
+The Nix manual is hosted on https://nixos.org/nix/manual.
+The underlying source files are located in [doc/manual/src](doc/manual/src).
+You can [use GitHub to edit these files](https://docs.github.com/en/repositories/working-with-files/managing-files/editing-files)
```

Comment:
Meh, not sure about that. People will find out anyway, but I wouldn't exactly encourage them to. I agree though it's easier for many people to just do it via GitHub, and that lowers the barrier for contributions. I don't know.

Reply:
Indeed. I also mentioned this because of the markdown preview.

Maybe it is good to mention this is only for small changes, but refer to the hacking page in the manual for bigger changes?

Reply:
Yes, sounds good.

Reply:
Done :+1:

I also squashes commits and rebased on master.